accountability.py

```diff
@@ -90,177 +114,268 @@ def get_page_values(job, step, condition):
+    return render_template('base.html', color='gray', page='waiting', reason='', img_ids=[], img_count=NUM_IMAGES)
 
-@app.route('/poll_submit')
+# Checking synchronicity of observer and moderator
+# Observer goes back to moderate
+# Moderator goes to done page, presents turk code to submit
+# TODO: keep this?
+@app.route('/' + POLL_SUBMIT_PAGE)
 def poll_submit():
     out_status = {}
     out_status['room'] = request.args.get('room')
     out_status['room_status'], out_status['observer'], out_status['moderator'] = query_db('select status, observer, moderator from submit_room_status where room=?', [out_status['room']], one=True)
     return jsonify(out_status)
 
-@app.route('/dashboard')
+@app.route('/' + DASHBOARD_PAGE)
 def dashboard():
     participants=query_db('select * from participants', one=False)
-    control = [p for p in participants if p[2]=='control']
-    experiment = [p for p in participants if p[2]=='experimental']
+    control = [p for p in participants if p[2]==CONDITION_CON_VAL]
+    experiment = [p for p in participants if p[2]==CONDITION_EXP_VAL]
     pairs = []
     for p in experiment:
         q = query_db('select mod_id from pairs where obs_id=?', [p[1]], one=True)
         if q is None:
             pairs.append((q,))
-        else: 
+        else:
             pairs.append(q)
-    
+
     exp_pairs = [a + b for a, b in zip(experiment, pairs)]
 
     control_html = ''.join(['<tr><th scope="row">{}</th><td>{}</td></tr>'.format(c[0], c[1]) for c in control])
     experiment_html = ''.join(['<tr><th scope="row">{}</th><td>{}</td><td>{}</td></tr>'.format(c[0], c[3], c[1]) for c in exp_pairs])
 
     return render_template('dashboard.html', control_html=control_html, experiment_html=experiment_html)
 
-@app.route("/submit_mods", methods=['POST'])
+@app.route("/" + OBS_TO_MOD_PAGE)
+def obs_to_mod():
+    turkId = session.get(TURK_ID_VAR)
+    job = JOB_MOD_VAL
+    condition = CONDITION_EXP_VAL
+
+    return redirect(url_for(WAIT_PAGE) + '?' + TURK_ID_VAR + '=' + str(turkId) + '&' + JOB_VAR + '=' + str(job) + '&' + CONDITION_VAR + '=' + str(condition))
+
+@app.route("/" + SUBMIT_MODS_PAGE, methods=['POST'])
 def accept_moderations():
-    json = request.get_json()
+    json = request.json
+
+    pair_id = json['pair_id']
+    img_ids = json['img_ids']
+    decisions = json['decisions']
+    for i in range(NUM_IMAGES):
+        if pair_id == 0:
+            query = 'insert into moderations(decision, img_id) VALUES(?, ?)'
+            query_db(query, [decisions[i], img_ids[i]], one=True)
+        else:
+            query = 'insert into moderations(decision, img_id, pair_id) VALUES(?, ?, ?)'
+            query_db(query, [decisions[i], img_ids[i], pair_id], one=True)
+
+    query_db('update pairs set mod_submitted=? where id=?', [True, pair_id])
 
-@app.route("/")
+    return jsonify(status='success');
+
+@app.route("/" + CHECK_MOD_SUBMITTED_PAGE, methods=['POST'])
```

Comment:
What is the rationale for splitting out the router pages (if I understood this right)? Will this still allow us to have a single URL for workers to be redirected to from the wait page?

Reply:
The router (root) page was changed from @app.route("/") to @app.route("/" + ROOT_PAGE) where ROOT_PAGE is an empty string, as seen on the new line 308. I did this in the refactoring step in the case that, for whatever reason, we might want to change the name of the routing page to something else. In terms of functionality, the old and new def index() functions work the exact same way, they serve as a single point for routing after being redirected from the wait page. Hopefully I understood your comment correctly and this covers the issue you brought up.